tb_plugin/torch_tb_profiler/profiler/trace.py

```diff
@@ -159,20 +177,26 @@ def create_event(event) -> Optional[BaseEvent]:
         raise
 
 
-def create_trace_event(event) -> Optional[BaseEvent]:
+def create_trace_event(event, is_pytorch_lightning) -> Optional[BaseEvent]:
     category = event.get('cat')
     event_type = EventTypeMap.get(category)
     if event_type == EventTypes.OPERATOR:
         name = event.get('name')
         if name and name.startswith('ProfilerStep#'):
             return ProfilerStepEvent(event)
-        else:
-            return OperatorEvent(event_type, event)
+        if is_pytorch_lightning:
+            if name and name.startswith('[pl][profile]'):
+                return PLProfileEvent(event)
+            elif name and name.startswith('[pl][module]'):
+                return PLModuleEvent(event)
+        return OperatorEvent(event_type, event)
     elif event_type == EventTypes.PYTHON:
         return OperatorEvent(event_type, event)
     elif event_type == EventTypes.KERNEL:
         return KernelEvent(event_type, event)
     elif event_type == EventTypes.PYTHON_FUNCTION:
```

Comment:
is the following more concise?

`elif event_type == EventTypes.PYTHON_FUNCTION and not is_pytorch_lightning: `

Reply:
The logic is different. If write like this and the is_pytorch_lightning is true, all PythonFunctionEvent will become DurationEvent and will fail with error like "DurationEvent has no 'Python id' args"...